app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -347,35 +396,22 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
     // POST IN CHANNEL
     // ========================
 
-    _createPostsInChannelRecord = (channelId: string, earliest: number, latest: number, prepareRecordsOnly = false): Promise<PostsInChannelModel[]> => {
-        // We should prepare instead of execute
-        if (prepareRecordsOnly) {
-            return this.prepareRecords({
-                tableName: POSTS_IN_CHANNEL,
-                createRaws: [{record: undefined, raw: {channel_id: channelId, earliest, latest}}],
-                transformer: transformPostsInChannelRecord,
-            });
-        }
-        return this.execute({
-            createRaws: [{record: undefined, raw: {channel_id: channelId, earliest, latest}}],
+    _createPostsInChannelRecord = async (channelId: string, earliest: number, latest: number, prepareRecordsOnly = false): Promise<PostsInChannelModel[]> => {
+        const models = await this.prepareRecords({
```

Comment:
Made several changes along this file to always prepare and then batch if we need to, instead of using two different functions with the same arguments over and over.